### ISSUE TITLE
Client disconnects every 4 hours

### ISSUE THREAD

**Issue by alessandrossi97** [dotnet/MQTTnet#1733]
### Describe your question
I have a C# console application in which a Mqtt client sends data to a server. The application is running with Docker on a remote PC with Ubuntu 22.04, SIM card and GSM module. In normal circumstances everything works fine but every 4 hours the event IMqttClient.DisconnectedAysnc is fired and I cannot understand why. When it happens, I try to ping google.com and the command is always successful, so I don't think it is a connection issue.
I have the same app running on a PC with the same specs and settings but with cabled connection to internet and is not suffering from this problem.
I don't know if this is related to the MQTTnet package or not.
Event handler function:

```
private async Task MqttClient_DisconnectedAsync(MqttClientDisconnectedEventArgs arg)
{
    try
    {
        _logger.Warning($"Mqtt client disconnected, reason: {arg.Reason}, ex: {arg.Exception?.Message}");
        Ping();
        await Connect();
    }
    catch (Exception ex)
    {
        _logger.Error(ex, "Mqtt client disconnected, reconnect failed.");
    }
}
```

Some logs:
```
2023-05-08 11:08:28.692 +00:00 [WRN] Mqtt client disconnected, reason: NormalDisconnection, ex: The operation has timed out.
2023-05-08 15:08:08.957 +00:00 [WRN] Mqtt client disconnected, reason: NormalDisconnection, ex: The operation has timed out.
2023-05-08 19:08:05.591 +00:00 [WRN] Mqtt client disconnected, reason: NormalDisconnection, ex: The operation has timed out.
2023-05-08 23:08:08.776 +00:00 [WRN] Mqtt client disconnected, reason: NormalDisconnection, ex: The operation has timed out.
```
Has anyone had this problem before?

Any help is really appreciated
Thanks,
Alessandro

### Which project is your question related to?

- Client

**Comment by logicaloud** [comment]
My first thought is that the GSM service may not like long lasting connections and causes the disconnect. The Ping may then still succeed because that simply establishes a new connection briefly and disconnects again. - Is it possible to test with a simple TCP server instead of the MQTT server? The client could then connect to the simple TCP server and send some keep-alive messages for 4 hours to see if it also disconnects.

Other than that, I would suggest to design around disconnects (persisted messages, reconnecting to the same session, etc., if required) because disconnects can happen for whatever reason; a quick reconnect every four hours is then not too bad.

**Comment by jiangzhuangxiansheng** [comment]
Yes, I also encountered the same situation, the same situation occurs in the sim environment
![image](https://github.com/dotnet/MQTTnet/assets/33273148/7b0b10d3-1016-4d27-8217-760811f0f2cc)

**Comment by WangDeBug** [comment]
我们也在3台PC部署了mqttclient通过4G网络发送到服务器，也差不多4小时左右就会触发事件 IMqttClient.DisconnectedAysnc。同样的应用程序在通过wifi和有线网络发送的时候没有此现象。

**Comment by antgraf** [comment]
Yes, I think it was reported earlier in #1084.
I also experience this. Client stops receiving data and is disconnected by the broker.
`1689714102: Client 8e86b6bd171b43c4a8666dd9d26ffeb9 has exceeded timeout, disconnecting.`
